src/components/Form/ServiceForm.tsx

```diff
@@ -174,99 +185,112 @@ function ServiceForm({ serviceId }: { serviceId?: string }) {
+                />
                 <span className='text-red-500'>
-                  <ErrorMessage name='rateToken' />
+                  <ErrorMessage name='about' />
                 </span>
               </label>
-            </div>
 
-            <SubmitButton isSubmitting={isSubmitting} label='Post' />
-          </div>
-        </Form>
-      )}
-    </Formik>
+              <label className='block'>
+                <span className='text-gray-700'>Keywords</span>
+
+                <SkillsInput
+                  initialValues={existingService?.description?.keywords_raw}
+                  entityId={'keywords'}
+                />
+
+                <Field type='hidden' id='keywords' name='keywords' />
+              </label>
+
+              <div className='flex'>
+                <label className='block flex-1 mr-4'>
+                  <span className='text-gray-700'>Amount</span>
+                  <Field
+                    type='number'
+                    id='rateAmount'
+                    name='rateAmount'
+                    className='mt-1 mb-1 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50'
+                    placeholder=''
+                  />
+                  <span className='text-red-500 mt-2'>
+                    <ErrorMessage name='rateAmount' />
+                  </span>
+                </label>
+                <label className='block'>
+                  <span className='text-gray-700'>Token</span>
+                  <Field
+                    component='select'
+                    id='rateToken'
+                    name='rateToken'
+                    className='mt-1 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50'
+                    placeholder=''
+                    onChange={(e: { target: { value: string } }) => {
+                      const token = allowedTokenList.find(
+                        token => token.address === e.target.value,
+                      );
+                      setSelectedToken(token);
+                      setFieldValue('rateToken', e.target.value);
+                    }}>
+                    <option value=''>Select a token</option>
+                    {allowedTokenList.map((token, index) => (
+                      <option key={index} value={token.address}>
+                        {token.symbol}
+                      </option>
+                    ))}
+                  </Field>
+                  <span className='text-red-500'>
+                    <ErrorMessage name='rateToken' />
+                  </span>
+                </label>
+              </div>
+
+              <SubmitButton isSubmitting={isSubmitting} label='Post' />
+            </div>
+          </Form>
+        )}
+      </Formik>
+      <Web3Mail showMailModal={showMailModal} setShowMailModal={setShowMailModal} />
```

Comment:
We need to work on a more global web3mail that could be integrated from anywhere much code on the parent component